core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -60,6 +60,10 @@ public static Catalog wrap(
     return new CachingCatalog(catalog, caseSensitive, expirationIntervalMillis);
   }
 
+  public boolean wrapped_is_instance(Class<?> cls) {
```

Comment:
this can be removed once #11093 gets in

Reply:
This is used for checking the type of the catalog impl that `CachingCatalog` is wrapping so that we can correctly respect the purge property at https://github.com/apache/iceberg/pull/11317/files#diff-bd61838d4e3a9aef52a670696750b30deac10b90f526435e32beacb0107eea24R376. Unless `CachingCatalog` is only used in tests, I think we need to keep this method?